common/dictionary-db/dictionary-db.ts

```diff
@@ -0,0 +1,1416 @@
+                    status: localTokenInput.status,
+                    lemmas: localTokenInput.lemmas,
+                    states: localTokenInput.states,
+                    cardIds: [],
+                });
+            }
+            return this._saveRecordBulk(records);
+        });
+    }
+
+    async deleteRecordLocalBulk(inputProfile: string | undefined, tokens: string[]): Promise<number> {
+        if (!tokens.length) return 0;
+        const profile = this._getProfile(inputProfile);
+        return this.db.tokens
+            .where('[token+source+track+profile]')
+            .anyOf(tokens.map((token) => [token, DictionaryTokenSource.LOCAL, LOCAL_TOKEN_TRACK, profile]))
+            .delete();
+    }
+
+    async deleteProfile(profile: string): Promise<[number, number, number]> {
+        return this.db.transaction('rw', this.db.meta, this.db.tokens, this.db.ankiCards, () =>
+            Promise.all([
+                this.db.meta.where('profile').equals(profile).delete(),
+                this.db.tokens.where('profile').equals(profile).delete(),
+                this.db.ankiCards.where('profile').equals(profile).delete(),
+            ])
+        );
+    }
+
+    private async _getFromSourceBulk(
+        profile: string,
+        track: number,
+        source: DictionaryTokenSource,
+        tokens: string[]
+    ): Promise<Map<string, DictionaryTokenRecord[]>> {
+        if (!tokens.length) return new Map();
+        return this.db.tokens
+            .where('[token+source+track+profile]')
+            .anyOf(tokens.map((token) => [token, source, track, profile]))
+            .toArray()
+            .then((records) => {
+                if (!records.length) return new Map();
+                const tokenRecordMap = new Map<string, DictionaryTokenRecord[]>();
+                for (const record of records) {
+                    const val = tokenRecordMap.get(record.token);
+                    if (val) val.push(record);
+                    else tokenRecordMap.set(record.token, [record]);
+                }
+                return tokenRecordMap;
+            });
+    }
+
+    private async _getAnkiCards(profile: string): Promise<DictionaryAnkiCardRecord[]> {
+        return this.db.ankiCards.where('profile').equals(profile).toArray();
+    }
+
+    private async _getAnkiCardsByNoteIdBulk(
+        profile: string,
+        noteIds: number[]
+    ): Promise<Map<number, DictionaryAnkiCardRecord[]>> {
+        if (!noteIds.length) return new Map();
+        return this.db.ankiCards
+            .where('[profile+noteId]')
+            .anyOf(noteIds.map((noteId) => [profile, noteId]))
+            .toArray()
+            .then((ankiCards) => {
+                if (!ankiCards.length) return new Map();
+                const noteIdRecordMap = new Map<number, DictionaryAnkiCardRecord[]>();
+                for (const ankiCard of ankiCards) {
+                    const val = noteIdRecordMap.get(ankiCard.noteId);
```

Comment:
Another note on variable naming - in general if you can make the code easier to read by just choosing a different name then that's probably a good choice (IMO). For example I think `cardRecords` would be better, especially since `noteIdRecordMap` doesn't make it completely obvious what the value of the map is.